test/UnitTest/Microsoft.AspNet.OData.Test.Shared/DeltaTest.cs

```diff
@@ -883,5 +1152,60 @@ public override int GetHashCode()
                 throw new NotImplementedException();
             }
         }
+
+        public class CustomerEntity
+        {
+            public int ID { get; set; }
+
+            public string FirstName { get; set; }
+
+            public string LastName { get; set; }
+
+            public AddressEntity Address { get; set; }
+        }
+
+        public class AddressEntity : IEquatable<AddressEntity>
+        {
+            public int ID { get; set; }
+
+            public string StreetAddress { get; set; }
+
+            public string City { get; set; }
+
+            public string State { get; set; }
+
+            public int ZipCode { get; set; }
+
+            public override bool Equals(object obj)
+            {
+                return Equals(obj as AddressEntity);
+            }
+
+            public bool Equals(AddressEntity other)
+            {
+                return other != null &&
```

Comment:
It might be good to have a reference check to: `return other != null && (other == this || (...));